sdks/java/io/common/src/test/java/org/apache/beam/sdk/io/common/IOTestPipelineOptions.java

```diff
@@ -100,4 +100,14 @@
   String getFilenamePrefix();
 
   void setFilenamePrefix(String prefix);
+
+  @Description("Google cloud storage - bucket_name/path")
+  String getGcsLocation();
```

Comment:
Why don't we use fileNamePrefix for all file-systems instead of introducing a property per file-system ?

Reply:
We can use `--filenamePrefix`, but then we need to provide full communication scheme there for GCS or HDFS, for instance `gs://bucket/path/file` or `hdfs://hadoop-master:port/dfs-path/file`. If we assume that user running tests will know it then those two gcsLocation and hdfsLocation could be ommited. This is basically implementation of our proposal https://docs.google.com/document/d/1dA-5s6OHiP_cz-NRAbwapoKF5MEC1wKps4A5tFbIPKE/edit#heading=h.29mfbxd6kc64 . Do you think would be better to remove those two pipeline options and just depend on pipelinePrefix ? Should I also remove protocol resolving part then ?

Reply:
We can use `--filenamePrefix`, but then we need to provide full communication scheme there for GCS or HDFS, for instance `gs://bucket/path/file` or `hdfs://hadoop-master:port/dfs-path/file`. If we assume that user running tests will know it then those two gcsLocation and hdfsLocation could be ommited. This is basically implementation of our proposal https://docs.google.com/document/d/1dA-5s6OHiP_cz-NRAbwapoKF5MEC1wKps4A5tFbIPKE/edit#heading=h.29mfbxd6kc64 . Do you think would be better to remove those two pipeline options and just depend on pipelinePrefix ? Should I also remove protocol resolving part then ?

Reply:
I think it makes sense to simplify and make 'fileNamePrefix' the full prefix. Later if we hit a case where this is inadequate we can revisit.